test/sql/compaction/merge_files_expired_snapshots.test

```diff
@@ -0,0 +1,128 @@
+# name: test/sql/compaction/merge_files_expired_snapshots.test
+# description: test ducklake merges files from expired snapshots if they also belong to current snapshot
+# group: [compaction]
+
+require ducklake
+
+require parquet
+
+statement ok
+ATTACH 'ducklake:__TEST_DIR__/ducklake_merge_expire_snapshots_schema.db' AS ducklake
+
+statement ok
+USE ducklake;
+
+statement ok
+CREATE TABLE example (key VARCHAR, value VARCHAR);
+
+statement ok
+INSERT INTO example (key, value) VALUES ('foo', 'bar');
+
+statement ok
+INSERT INTO example (key, value) VALUES ('baz', 'qux');
+
+query II
+SELECT snapshot_id, changes FROM snapshots();
+----
+0	{schemas_created=[main]}
+1	{tables_created=[main.example]}
+2	{tables_inserted_into=[1]}
+3	{tables_inserted_into=[1]}
+
+query I
+SELECT count(*) FROM ducklake_list_files('ducklake', 'example');
+----
+2
+
+statement ok
+CALL ducklake_expire_snapshots('ducklake', versions => [2]);
```

Comment:
This still won't work if we expire `versions => [1, 2]` because then we have lost the schema version. I think the only way to fix this is to track the schema version outside of the `ducklake_snapshot` table, for example we can add this table:


```
CREATE TABLE ducklake_schema_versions(begin_snapshot BIGINT, schema_version BIGINT);
```

This table can be populated whenever `SchemaChangesMade()` is true with the new snapshot id and the new schema version. 

For migration, we could do e.g.:

```
INSERT INTO ducklake_schema_versions SELECT MIN(snapshot_id), schema_version FROM ducklake_snapshot GROUP BY schema_version ORDER BY schema_version;
```

Then we can figure out the schema version associated with a file also after snapshots have been expired.